### PR TITLE
[ONNX] Use ov::v12::ScatterElementsUpdate in ONNX FE ScatterElements (enable reduction)

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 - Use ov::v12::ScatterElementsUpdate in ONNX FE ScatterElements (enable reduction)
 - Fix ScatterElementsUpdate reference implementation (add indices normalization if reduction is "none")

### Tickets:
 - 111107

### CODE REVIEW

**Review by mbencer — APPROVED** [review]
LGTM, btw it looks like you enabled on the occasion 3 ONNX compliance tests (already xfailed)

**Review by mmikolajcz — APPROVED** [review]

**Comment by mitruska** [comment]
Added more tests and required fix to reference implementation (indices normalization if reduction is "none").
Please take a look @mbencer, @mmikolajcz.

**Review by mbencer — APPROVED** [review]

**Review by mmikolajcz — APPROVED** [review]

File: src/frontends/onnx/tests/runtime/ie/unit_test.manifest [hunk 1]
```diff
@@ -412,3 +412,6 @@ IE_CPU.onnx_model_top_k_repeating_unsorted
 
 # Accuracy regression - Ticket 105909
 IE_CPU.onnx_model_attention_qkv_hidden_sizes
+
+# Ticket: 111110, ScatterElementsUpdate-12 (not fully implemented)
+IE_CPU.onnx_model_scatter_elements_none_neg_ind_opset18
```
Comment (rkazants) [inline src/frontends/onnx/tests/runtime/ie/unit_test.manifest:417]: do we have downgrade transformation for plugins that do not support new version?

Comment (mitruska) [reply src/frontends/onnx/tests/runtime/ie/unit_test.manifest:417]: Of course, it is already on master: https://github.com/openvinotoolkit/openvino/blob/feab0ab75e9c4a567091461d70de4cf3eb4686c3/src/common/transformations/src/transformations/op_conversions/convert_scatter_elements_update12_downgrade.cpp#L13

Comment (mlukasze) [reply src/frontends/onnx/tests/runtime/ie/unit_test.manifest:417]: > do we have downgrade transformation for plugins that do not support new version?

yes

**Pull request merged**